icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -1113,12 +1113,23 @@ CollationBuilder::addWithClosure(const UnicodeString &nfdPrefix, const UnicodeSt
     return ce32;
 }
 
+// ICU-22517
+// This constant define a soft limit for the addOnlyClosure to return
+// error to avoid slow return due to unnecessary closure expansion.
+// Please let us know if you have a reasonable use case that needed
+// for a practical Collation rule that need to increase this limit.
+// This value is needed for compile the rule with eight Hanguls such as
+// "&a=b쫊쫊쫊쫊쫊쫊쫊쫊" without error, which should be more than realistic
+// usage.
```

Comment:
```suggestion
// This constant defines a soft limit for the addOnlyClosure to return an
// error, to avoid taking a long time for canonical closure expansion.
// Please let us know if you have a reasonable use case
// for a practical Collation rule that needs to increase this limit.
// This value is needed for compiling a rule with eight Hangul syllables such as
// "&a=b쫊쫊쫊쫊쫊쫊쫊쫊" without error, which should be more than realistic
// usage.
```